lib/contracts/feeoraclev2/feeparams.go

```diff
@@ -0,0 +1,141 @@
+package feeoraclev2
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/evmchain"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/tokens"
+	"github.com/omni-network/omni/monitor/xfeemngr"
+
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func feeParams(ctx context.Context, srcChainID uint64, destChainIDs []uint64, backends ethbackend.Backends, pricer tokens.Pricer,
+) ([]bindings.IFeeOracleV2FeeParams, error) {
+	// used cached pricer, to avoid multiple price requests for same token
+	pricer = tokens.NewCachedPricer(pricer)
+
+	srcChain, ok := evmchain.MetadataByID(srcChainID)
+	if !ok {
+		return nil, errors.New("meta by chain id", "chain_id", srcChainID)
+	}
+
+	var resp []bindings.IFeeOracleV2FeeParams
+	for _, destChainID := range destChainIDs {
```

Comment:
We might also want to track srcChainId fee params on the source. Litle weird,  but we do that right now. And I know there was a reason. I'll go find if we still need to

Reply:
All chainIDs are now passed in instead of filtering out the source